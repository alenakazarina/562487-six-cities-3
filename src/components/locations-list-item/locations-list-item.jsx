import React from 'react';
import {string, bool, func} from 'prop-types';

const LocationsListItem = ({city, isActive, onTabClick}) => {
  const activeClass = isActive ? `tabs__item--active` : ``;
  return (
    <li className="locations__item" key={city}>
      <a
        className={`locations__item-link tabs__item ${activeClass}`}
        onClick={() => onTabClick(city)}
      >
        <span>{city}</span>
      </a>
    </li>
  );
};

LocationsListItem.propTypes = {
  city: string.isRequired,
  isActive: bool.isRequired,
  onTabClick: func.isRequired
};

export default React.memo(LocationsListItem);
