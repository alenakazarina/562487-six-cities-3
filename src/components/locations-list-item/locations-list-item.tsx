import * as React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import {ActionCreator as OffersActionCreator} from '../../reducers/offers/offers';
import {AppRoute} from '../../const';

interface Props {
  nodeType: `li` | `div`;
  city: string;
  isActive: boolean;
  setActiveCity: () => void;
}

const LocationsListItem: React.FC<Props> = (props: Props) => {
  const {
    nodeType,
    city,
    isActive,
    setActiveCity
  } = props;
  const activeClass = isActive ? `tabs__item--active` : ``;

  return nodeType === `li` ? (
    <li className="locations__item" key={city}>
      <a className={`locations__item-link tabs__item ${activeClass}`}
        onClick={setActiveCity}
      >
        <span>{city}</span>
      </a>
    </li>
  ) : (
    <div className="locations__item" key={city}>
      <Link to={AppRoute.ROOT} className={`locations__item-link tabs__item ${activeClass}`}
        onClick={setActiveCity}
      >
        <span>{city}</span>
      </Link>
    </div>
  );
};

const mapDispatchToProps = (dispatch, props) => ({
  setActiveCity({currentTarget}) {
    const city = currentTarget.text;
    if (props.isActive) {
      return;
    }
    dispatch(OffersActionCreator.setActiveCity(city));
  }
});

export {LocationsListItem};
export default connect(null, mapDispatchToProps)(LocationsListItem);
