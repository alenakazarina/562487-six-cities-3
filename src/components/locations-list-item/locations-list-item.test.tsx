import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import {LocationsListItem} from './locations-list-item';

const CITY = `Amsterdam`;
const mockFn = jest.fn();

describe(`LocationsListItem`, () => {
  it(`should render LocationsListItem as listItem with active class`, () => {
    const tree = renderer.create(
        <LocationsListItem
          nodeType="li"
          city={CITY}
          isActive={true}
          onLocationsItemClick={mockFn}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`should render LocationsListItem as div`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <LocationsListItem
            nodeType="div"
            city={CITY}
            isActive={false}
            onLocationsItemClick={mockFn}
          />
        </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
