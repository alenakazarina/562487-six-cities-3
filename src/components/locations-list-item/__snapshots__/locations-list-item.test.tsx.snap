// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationsListItem should render LocationsListItem as div 1`] = `
<div
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item "
    href="/"
    onClick={[Function]}
  >
    <span>
      Amsterdam
    </span>
  </a>
</div>
`;

exports[`LocationsListItem should render LocationsListItem as listItem with active class 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item tabs__item--active"
    onClick={[MockFunction]}
  >
    <span>
      Amsterdam
    </span>
  </a>
</li>
`;
